### ISSUE TITLE
Testing stage

### ISSUE THREAD

**Issue by Nawaffarhan** [quanghuy0497/Mask_R-CNN#2]
I am confused in training part using 28x28 for compute loss function. In the evaluation or testing part to compute AP used the 28x28 after that up sampling with therthold 0.5 or upsampling to compute overlap??

**Comment by quanghuy0497** [comment]
Your question to me is a little confusing, so I encourage you should go to the original repository Matterport/Mask_RCNN looking for the answer. If you want to imply the threshold of 0.5 when computing AP, that is just the IoU threshold to determine if the predicted mask/bounding box is correct with the ground truth (see how AP is computed for further detail)

**Comment by Nawaffarhan** [comment]
My question is when evaluate iou mask for evaluate AP with different
therthold.
 The predict mask is 28x28 to evaluate AP or resize to original size of roi
to evaluate AP

**Comment by quanghuy0497** [comment]
I believe the predicted mask is evaluated AP with the size of 28x28.

Vào 20:32, CN, 1 thg 5, 2022 Nawaffarhan ***@***.***> đã viết:

**Comment by Nawaffarhan** [comment]
Thank you for the clarification.

**Comment by Nawaffarhan** [comment]
Please I has questions could you please help me.

The question is if used 1024x1024 image size and finally has RoI is 8x8

and fixed mask feature is 14x14, how could convert small size of feature
map to bigger size?

**Comment by quanghuy0497** [comment]
Dear Nawaffahan,

For **_"downsampling"_** a feature matrix (e.g. from 512x512 to 32x32) we normally use **Convolutional layer** with Convolutional Operation, which is a basic of CNN.

On the otherhand, for **_"upsampling"_** a feature matrix (e.g. from 7x7 to 14x14), we can use the **Deconvolutional Layer**, which is an inverse to Convolutional.

This deConv layer/operation is quite popular in Segmentation architectures. You can read the papers about Fully Convolutional Network or U-Net for better intuition.

In case you have any further question, please read the original paper about Mask R-CNN and comparing it with the implementation code beforehand.

Best,

**Comment by Nawaffarhan** [comment]
My question the small object in coco where less than 32x32.
 And im mask rcnn the scale ration is 4 for 256 and roi align used convert
the region to 14x14. My question is if I have small object (28x28) will be
28/4 is 7 how to roi align will converted to fixed size which is 14x14 and
I read the code of maskrcnn don't used upsampling in this case.
